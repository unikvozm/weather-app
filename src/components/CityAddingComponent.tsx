import * as React from 'react';
import {TextInput} from 'react-native-paper';
import {View, Button, StyleSheet} from 'react-native';

type Props = {
  addCity: (city: string) => void;
  closeAddingMode: () => void;
};

export const CityAddingComponent = ({addCity, closeAddingMode}: Props) => {
  const [query, setQuery] = React.useState<string>('');
  const onSubmit = () => {
    addCity(query);
    closeAddingMode();
  };
  return (
    <View>
      <TextInput
        label="City"
        value={query}
        onChangeText={(text) => setQuery(text)}
        placeholder="Start typing a city"
      />
      <View style={styles.btnsContainer}>
        <Button onPress={onSubmit} title="Add" disabled={!query} />
        <Button onPress={closeAddingMode} title="Cancel" />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  btnsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    margin: 10,
    flexShrink: 3,
  },
});
