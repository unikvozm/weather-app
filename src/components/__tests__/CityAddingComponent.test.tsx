import React from 'react';
import renderer from 'react-test-renderer';
import {CityAddingComponent} from '../CityAddingComponent';

test('CityAddingComponent renders correctly', () => {
  const tree = renderer
    .create(<CityAddingComponent addCity={jest.fn} closeAddingMode={jest.fn} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
