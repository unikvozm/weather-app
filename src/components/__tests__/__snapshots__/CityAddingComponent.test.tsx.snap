// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityAddingComponent renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Object {
          "minHeight": 64,
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <Text
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          City
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          City
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "textAlign": "left",
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlignVertical": "center",
            },
            Array [
              Object {},
            ],
          ]
        }
        testID="City to add"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexShrink": 3,
        "justifyContent": "space-evenly",
        "margin": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
